### PR TITLE
Comparing date math to years doesn't work as expected

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

**Comment by eanders** [comment]
@awisema, I'm not finding any other similar bugs in the codebase (but I did look)

File: app/models/concerns/hud_reports/clients.rb [hunk 1]
```diff
@@ -79,7 +79,7 @@ module HudReports::Clients
 
     private def annual_assessment_expected?(enrollment)
       return false if enrollment.last_date_in_program.present? &&
-        enrollment.last_date_in_program - enrollment.first_date_in_program < 1.year
+        enrollment.last_date_in_program - enrollment.first_date_in_program < 365
```
Comment (awisema) [inline app/models/concerns/hud_reports/clients.rb]: On calculating anniversary dates, HUD says:

_It is important to use the “age” method of determining client anniversaries due to leap years; using “one year = 365 days” will eventually incorrectly offset the calculated anniversaries of long-term stayers._

Maybe
`return false if enrollment.first_date_in_program + 1.year > enrollment.last_date_in_program`?

Comment (eanders) [reply app/models/concerns/hud_reports/clients.rb]: That's the ticket!

Comment (eanders) [reply app/models/concerns/hud_reports/clients.rb]: Though in this case, we're just checking to see if it's been at least a year, so 365 will be wrong once every 4 years, but it won't get compounded.

**Comment by eanders** [comment]
@awisema, I believe the logic now is correct and much easier to read

File: app/models/concerns/hud_reports/clients.rb [hunk 2]
```diff
@@ -77,11 +77,16 @@ module HudReports::Clients
...
+    # If the enrollment lasted more than one year
+    # and the reporting period end is more than a year since the beginning of the enrollment
+    # and the enrollment started more than one year ago
     private def annual_assessment_expected?(enrollment)
-      return false if enrollment.last_date_in_program.present? &&
-        enrollment.last_date_in_program - enrollment.first_date_in_program < 365
+      return false unless enrollment.head_of_household?
 
-      enrollment.head_of_household? && enrollment.first_date_in_program + 1.years < report_end_date
+      end_date = [enrollment.last_date_in_program, report_end_date, Date.current].compact.min
```
Comment (awisema) [inline app/models/concerns/hud_reports/clients.rb:88]: I'm worried about if people run a report with a report end date in the future...

`end_date = [enrollment.last_date_in_program, report_end_date].compact.min || Date.current`?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/concerns/hud_reports/clients.rb (hunk 2 above)
Comment (awisema) [reply app/models/concerns/hud_reports/clients.rb:88]: Never mind... That would be pointless... `report_end_date` will never be nil.

Comment (awisema) [reply app/models/concerns/hud_reports/clients.rb:88]: So, `end_date = [enrollment.last_date_in_program, [report_end_date, Date.current].max].compact.min`>

**Review by awisema — APPROVED** [review]

**Pull request merged**